app/views/components/inputDate.scala.html

```diff
@@ -79,5 +79,11 @@
             )
         ),
         hint = hintText.map(hint => Hint(content = Text(messages(hint)))),
-        errorMessage = form(id).error.map(err => ErrorMessage(content = Text(messages(err.message, err.args: _*))))
+        errorMessage = {
+            form(id).error
+                .orElse(form(s"$id.day").error)
+                .orElse(form(s"$id.month").error)
+                .orElse(form(s"$id.year").error)
+                .map(err => ErrorMessage(content = Text(messages(err.message, err.args: _*)), visuallyHiddenText = Some(messages("site.error"))))
```

Comment:
Better to format it to keep on multiple line (as below) to increase the readability 

 errorMessage = {
            form(id).error
                .orElse(form(s"$id.day").error)
                .orElse(form(s"$id.month").error)
                .orElse(form(s"$id.year").error)
                .map(err => ErrorMessage(
                        content = Text(messages(err.message, err.args: _*)),
                        visuallyHiddenText = Some(messages("site.error"))
                        )
                    )
        }

Note: - If its showing in your IDE as above (on multiple lines) then ignore this comment